client/src/main/java/org/eclipse/hono/client/impl/CredentialsClientImpl.java

```diff
@@ -180,22 +203,28 @@ public static final String getTargetAddress(final String tenantId) {
                 .put(CredentialsConstants.FIELD_TYPE, type)
                 .put(CredentialsConstants.FIELD_AUTH_ID, authId)
                 .mergeIn(clientContext);
+        final TriTuple<CredentialsConstants.CredentialsAction, String, Integer> key = TriTuple
+                .of(CredentialsConstants.CredentialsAction.get, String.format("%s-%s", type, authId), clientContext.hashCode());
```

Comment:
what if client context is `null`?

In this context: can you also please fix the @throws tag of the JavaDoc of `CredentialsClient.get(String, String, JsonObject)` accordingly?

Reply:
In javadoc  of `CredentialsClient.get(String, String, JsonObject)` it is mentioned that `@param clientContext Optional bag of properties that can be used to identify the device`. Since it is optional, I thing it is valid to set as also null.

May be it makes more sense: if client context is not `null`, then pass its `hashcode` as the third argument, else pass `null` instead. WDYT?

Reply:
I was just hoping for you to fix it like this:
> @throws NullPointerException if any of the parameters other than client context are {@code null}

Reply:
> @throws NullPointerException if any of the parameters other than client context are {@code null}

Sorry, I didn't get the intention behind the JavaDoc statement. In the current code (in PR), if the client context is `null`, then the `NullPointerException` is thrown due to `clientContext.hashCode()`. Then the above JavaDoc statement is contradictory.  If we fix the code with ` Optional.ofNullable(clientContext).map(cc-> cc.hashCode()).orElse(null);`, then the above JavaDoc statement is valid and be used. 

If we intend to leave the code as it is in this PR, then no need to change any JavaDoc statement. In that case, the below is valid. 
```
     * @throws NullPointerException if any of the parameters is {@code null}.
     * @see RequestResponseClient#setRequestTimeout(long)
     */
    Future<CredentialsObject> get(String type, String authId, JsonObject clientContext);
```

Reply:
Yeah, I see what you mean. I misread `(except spanContext)` for `(except clientContext)`. So, it seems that the methods indeed require a non-null `JsonObject` as client context. However, the JSON may be empty.
So, the client context object itself should indeed never be `null`. However, I would prefer if we could add an explicit check at the top of the method then, instead of just running into a `NullPointerException` down here, ok?

Reply:
I have added an explicit check at the top.